File: Java/src/com/company/WarehouseCell.java
package com.company;

import java.util.concurrent.ExecutionException;

public class WarehouseCell {
    public int Id;
    public Item CellItem;
    private Integer ItemCount;
    public void setItemCount(int value){
        if (value > this.CellItem.MaxCapacity){
            ItemCount = value;
        } else {
            throw new ArithmeticException("");
        }
    }
    public int getItemCount(){
        return this.ItemCount;
    }

    public WarehouseCell() { }

    public WarehouseCell(int id, Item cellItem, Integer itemCount)
    {
        Id = id;
        CellItem = cellItem;
        ItemCount = itemCount;
    }
}

File: C#/ClassLibrary/WarehouseCell.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ClassLibrary
{
    public class WarehouseCell
    {
        public int Id { get; set; }
        public Item? CellItem { get; set; }
        public int? ItemCount 
        {
            set {
                if (value > this.CellItem.MaxCapacity)
                {
                    ItemCount = value;
                }
                else
                {
                    throw new Exception("out of storage exception");
                }
            }
            get { return ItemCount; } 
        }

        public WarehouseCell(int id, Item cellItem, int? itemCount)
        {
            Id = id;
            CellItem = cellItem;
            ItemCount = itemCount;
        }
    }
}
